dwave_networkx/algorithms/matching.py

```diff
@@ -13,10 +13,164 @@
 #    limitations under the License.
 
 import itertools
+import numbers
+import warnings
+
+import dimod
+import networkx as nx
 
 from dwave_networkx.utils import binary_quadratic_model_sampler
 
-__all__ = ['min_maximal_matching', 'is_matching', 'is_maximal_matching']
+__all__ = ['is_matching',
+           'is_maximal_matching',
+           'matching_bqm',
+           'maximal_matching_bqm',
+           'min_maximal_matching',
+           'min_maximal_matching_bqm',
+           ]
+
+
+def matching_bqm(G):
+    """Return a BQM with ground states corresponding to the possible matchings
+    of G.
+
+    A matching is a subset of edges in which no node occurs more than
+    once.
+
+    Finding valid matchings can be done is polynomial time, so this BQM
+    is only efficient when combined with other constraints and objectives.
+
```

Comment:
efficient or useful?

polynomial time. This BQM
may be useful when combined with other constraints and objectives
to solve a larger hard problem.

Reply:
I think we need to connect those two concepts a bit more explicitly, how about 
```    
    Finding valid matchings can be done in polynomial time, so finding matching
    with BQMs is generally inefficient.
    This BQM may be useful when combined with other constraints and objectives.
```